### DISCUSSION TITLE
Can we install global packages from github?

### DISCUSSION THREAD

**Discussion by Shobhit0109** [prefix-dev/pixi#2152] [Q&A]
As, we can add pacakges in a project directly from github. Is there any way to install globally?

**Comment by rbhanot4739** [comment]
I would also like that. I mainly use pixi to install/manage tools globally; however, as of now there are certain packages that are not available in the conda-forge channels, so I have to install those packages separately via some other tools, few common ones for me are tmuxinator and [tealdeer|tllr-replacement-in-rust](https://github.com/tealdeer-rs/tealdeer). 

So it would be much nicer to be able to install packages directly from github.

**Comment by wolfv** [comment]
Hi all, there is currently no way to do this directly, but it's always possible to build software either from scratch or by repackaging pre-built software. 

For tealdeer we could (for example) easily make a package on rust-forge: https://github.com/wolfv/rust-forge - I added a recipe for tealdeer here: https://github.com/wolfv/rust-forge/commit/2dadefb7b3749b43e90946595d7cfdd0657ae449. You will be able to install this with `pixi global install tealdeer -c https://prefix.dev/rust-forge`.

We could also contribute that to conda-forge.

**Reply by rbhanot4739** [reply]
@wolfv Awesome, thank you for the quick response and porting the package as well. Really appreciate that, however it didn't work as such.

```
pixi global install tealdeer -c https://prefix.dev/rust-forge
  × Couldn't install tealdeer
  ├─▶ failed to solve environment
  ╰─▶ Cannot solve the request because of: No candidates were found for tealdeer *.
```
Do i need to something additional as well in order to get this to work.

**Reply by wolfv** [reply]
I made a few mistakes. Packages are being uploaded right now! :)

**Reply by rbhanot4739** [reply]
Sure, thank you, so what is the general guidance for additional packages ?

**Reply by wolfv** [reply]
General guidance would be to add packages to `conda-forge`. That ensure the widest distribution. I can help!

**Reply by wolfv** [reply]
But if you re-run the command now, you should be able to get it from rust-forge in the meantime.

**Reply by rbhanot4739** [reply]
```pixi global install tealdeer -c https://prefix.dev/rust-forge
  × Couldn't install tealdeer
  ├─▶ failed to solve environment
  ╰─▶ Cannot solve the request because of: tealdeer * cannot be installed because there are no viable options:
      └─ tealdeer 1.7.0 would require
         └─ libgcc >=14, for which no candidates were found.```
Do I have to resolve dependencies manually, ideally it should handle dependencies wouldn't it ?

> General guidance would be to add packages to conda-forge. That ensure the widest distribution. I can help!

Thank you, do you want packages to be reported to you so that you can add/update to conda-forge or you want end users to repackage and add to conda-forge ?

**Reply by wolfv** [reply]
Argh. Try `-c rust-forge -c conda-forge`. Sorry! :)

**Reply by wolfv** [reply]
Ideally end-users start contributing themselves to conda-forge, but we are happy to help :)

**Reply by rbhanot4739** [reply]
Hey sorry to nag about it, but it still fails
```
pixi global install tealdeer -c rust-forge -c conda-forge
ERROR error=repodata not found
ERROR error=repodata not found
ERROR error=could not find subdir 'noarch' in channel 'https://conda.anaconda.org/rust-forge/'
  × Couldn't install tealdeer
  ├─▶ could not find subdir 'noarch' in channel 'https://conda.anaconda.org/rust-forge/'
  ╰─▶ HTTP status client error (404 Not Found) for url (https://conda.anaconda.org/rust-forge/noarch/repodata.json)
```

**Reply by rbhanot4739** [reply]
@wolfv Still not able to install the package

```
 pixi global install tealdeer
  × Couldn't install tealdeer
  ├─▶ failed to solve environment
  ╰─▶ Cannot solve the request because of: No candidates were found for tealdeer *.


❯ pixi global install tealdeer -c rust-forge -c conda-forge
ERROR error=repodata not found
ERROR error=could not find subdir 'noarch' in channel 'https://conda.anaconda.org/rust-forge/'
  × Couldn't install tealdeer
  ├─▶ could not find subdir 'noarch' in channel 'https://conda.anaconda.org/rust-forge/'
  ╰─▶ HTTP status client error (404 Not Found) for url (https://conda.anaconda.org/rust-forge/noarch/repodata.json)
```

Also I just to happen to notice that yazi (another one of popular rust tools) has version 2.4.0 available in rust-forge with 3.3 being the latest.
